### PR TITLE
Week 2 assessment jose salas

### PR DESCRIPTION
**Pull request by salas89jl**
# Submitting Week 2 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [x] Coding Challenges
- [x] Coding Concepts
- [x] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -1,59 +1,290 @@
...
+//  it("Takes a number inside the object and checks if its evenly divided by three (number % 3) or not (!)", () => {
+    
+//     const object1 = { number: 15 }
+//         // Expected output: "15 is divisible by three"
+//     const object2 = { number: 0 }  
+//         // Expected output: "0 is divisible by three"
+//     const object3 = { number: -7 }
+//         // Expected output: "-7 is not divisible by three"
+ 
+// expect(doesThatFit(object1)).toBe(true)
```
Comment (Genemichael25) [inline code-challenges.test.js:110]: Want to make sure our test expects the expected output, in this case the number being divisible by three or not in a string.

File: code-challenges.test.js [hunk 2]
```diff
@@ -1,59 +1,290 @@
...
+
+      at Object.expect (code-challenges.test.js:106:5)
+*/
 
 // b) Create the function that makes the test pass.
+/*
+Pseudo Code
+Create a fuction called doesThatFit
+Function needs to check if the object contains a "number" as property
+If property exist within the object, then it will check if "number" property has a typeof number as a value
```
Comment (Genemichael25) [inline code-challenges.test.js:143]: It's great that you are thinking of edge cases and a couple of other things you want to check for in your function. I would encourage you to get a working function, and then attempt to refactor.

File: code-challenges.test.js [hunk 3]
```diff
@@ -1,59 +1,290 @@
...
+
+    at Object.expect (code-challenges.test.js:167:9)
+*/
 // b) Create the function that makes the test pass.
 
+// Pseudo Code
+// Create a function named smallToBig
+// Input: an array of words
+// output: an array of the same words but all capilized.
+// hof = to .map
```
Comment (Genemichael25) [inline code-challenges.test.js:196]: Great job getting the expected output but this pseudocode needs a bit more. After you use .map, what do you intend to do and how?

File: code-challenges.test.js [hunk 4]
```diff
@@ -1,59 +1,290 @@
...
+          |       ^
+      226 |       expect(whereRU("academy")).toEqual("0")
+      227 |       expect(whereRu("challenges")).toEqual("2") */
+
+
 // b) Create the function that makes the test pass.
+// Create a function that takes in a string and logs the index of the first vowel.
+
+//Pseudo Code
+// 1. Create a function named whereRU that takes a string as an argument
```
Comment (Genemichael25) [inline code-challenges.test.js:261]: I like this approach! Is there a reason why you decided to have your pseudocode in between your function?

File: coding-concepts.js [hunk 5]
```diff
@@ -8,36 +8,43 @@
...
 
 const cohort = "Bravo 2023"
-// console.log(cohort.split(""))
 
-// a) Your answer:
-// b) Verify and explain:
+console.log(cohort.split("")) // --> ['B', 'r', 'a', 'v','o', ' ', '2', '0','2', '3']
+
+// a) Your answer: Before: This will log "Bravo 2023" as an array each character as value.
+// b) Verify and explain: After: I was correct, but I forgot that the spaces in the strings are also treated as a character/value
```
Comment (Genemichael25) [inline coding-concepts.js:15]: Great follow up!

File: coding-concepts.js [hunk 6]
```diff
@@ -8,36 +8,43 @@
...
+
 const greeter = (name) => {
-  `Hello, ${name}!`
+ return `Hello, ${name}!`
 }
-// console.log(greeter("LEARN Student"))
-
-// a) Your answer:
-// b) Verify and explain:
+console.log(greeter("LEARN Student")) // --> Hello, LEARN Student!
```
Comment (Genemichael25) [inline coding-concepts.js:30]: Great job verifying your explanation! 💯

**Review by Genemichael25 — CHANGES_REQUESTED** [review]
Resolved.

File: interview-questions.md [hunk 7]
```diff
@@ -9,49 +9,148 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+Researched answer: From OpenAI(ChatGPT4)
+    Parameter: Is a value that is passed into a function when the fuction is being called.
+    Argument: Is the actual value that is passed to the function when it its being called.
+
+```js
+// num1 and num2 are the parameters
+const addNumbers = (num1, num2) => {
+    return num1 + num2
+}
+// 5 and 10 are the actual values which are called arguments
```
Comment (Genemichael25) [inline interview-questions.md:23]: Love the use of examples!

File: interview-questions.md [hunk 8]
```diff
@@ -9,49 +9,148 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+4. What is iteration?
 
 Your answer:
+    Iteration is the process of repeating a set of instruction over and over again until certian(s) condition(s) are met.
+
+Researched answer: From OpenAI(ChatGPT4)
+    Iteration refers to the process of performing repeatedly a set of operations or instructions on a sequence of elements or data. Iteration is a fundimental concept in computer programming that allows programmers to execute a block of code over and over for a number of specified times or until a condition is met. 
+    
+    In JavaScript, it is commonly performed using loops, such as the for loop, while loop, and do-while loop. With these loops a programmer can repeatedly execute a block of code while a certain condition is true, or until completing a specified number iterations.
+
```
Comment (Genemichael25) [inline interview-questions.md:94]: Great, love that you mentioned for loops. .map is another example of an iterative method.

File: interview-questions.md [hunk 9]
```diff
@@ -9,49 +9,148 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 6. STRETCH: What is hoisting in JavaScript?
 
-Your answer:
+Your answer: 
+    I dont know, but I will most definitly look up the answer! My best educated guess is that it is referring to the programming language a platform, such as react, is based on.
+
+Researched answer: From OpenAI(ChatGPT4)
+    And I was way off target by a thousand miles!
+    Hoisting in Javascript is a behavior that allows the declaration of variables and functions to moved to the top of thier respective scopes during the compilation phase.
```
Comment (Genemichael25) [inline interview-questions.md:119]: Awesome job looking into this! 💯

File: code-challenges.test.js [hunk 10]
```diff
@@ -1,59 +1,290 @@
...
+Pseudo Code
+Create a fuction called doesThatFit
+Function needs to check if the object contains a "number" as property
+If property exist within the object, then it will check if "number" property has a typeof number as a value
+Then it will take that value and check if its % 3 === 0
+*/
+
+const doesThatFit = (object) => {
+   return object.number % 3 === 0
+     }
```
Comment (Genemichael25) [inline code-challenges.test.js:149]: I would give this one another try because you are definitely on the right path.

**Review by Genemichael25 — APPROVED** [review]
This is another solid performance on these assessments Jose! I love the way you set up your test for the # 2 code challenge and would like for you to do that for all the tests. Continue to work on your pseudocode and I think you have the tools necessary to finish # 1 in code challenges. This currently displays a novice level of comprehension but you are really close to Advanced Beginner. If you have any questions about any of the comments I left, please feel free to message me. Keep up the good work! 👏🏽
